var/spack/repos/builtin/packages/graph500/package.py

```diff
@@ -0,0 +1,50 @@
+##############################################################################
+# Copyright (c) 2013-2017, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/spack/spack
+# Please also see the NOTICE and LICENSE files for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Graph500(MakefilePackage):
+    """Graph500 reference implementations."""
+
+    homepage = "https://graph500.org"
+    url      = "https://github.com/graph500/graph500/archive/graph500-3.0.0.tar.gz"
+
+    version('3.0.0', 'a2ebb4783b21e2d86387a217776395e3')
+
+    depends_on('mpi@2.0:')
+
+    build_directory = 'src'
+
+    def eidt(self):
+        edit = FileFilter('src/Makefile')
+        edit.filter(r'^MPICC\s*=.*', 'MPICC={0}'.format(spec['mpi'].mpicc))
+
+    def install(self, spec, prefix):
+        with working_dir('src'):
```

Comment:
You can replace `'src'` with `self.build_directory` so you don't need to use as many ["magic numbers"](https://en.wikipedia.org/wiki/Magic_number_(programming))

Reply:
Fixed.